apache/datadog_checks/apache/apache.py

```diff
@@ -141,3 +163,19 @@ def _submit_metadata(self, value):
         version_parts = {name: part for name, part in zip(('major', 'minor', 'patch'), version.split('.'))}
         self.set_metadata('version', version, scheme='parts', final_scheme='semver', part_map=version_parts)
         self.log.debug("found apache version %s", version)
+
+    def _submit_scoreboard(self, value, tags):
+        max_workers = 0
+        metrics = {}
+        for _key, metric in self.SCOREBOARD_KEYS.items():
+            metrics[metric] = 0
+
+        for i in value:
+            max_workers += 1
+            metrics[self.SCOREBOARD_KEYS[i]] += 1
+
+        for _key, metric in self.SCOREBOARD_KEYS.items():
+            self.gauge(metric, metrics[metric], tags=tags)
```

Comment:
Small implementation nit, we can replace that code by the following one which is easier to understand

```suggestion
        for _key, metric in self.SCOREBOARD_KEYS.items():
            self.gauge(metric, value.count(_key), tags=tags)
        
        self.gauge(self.GAUGES['MaxWorkers'], len(value), tags=tags)
        self.log.debug("Scoreboard: %s", value)
```

Additionally we might want to add a validation check at the beginning of that method to make sure that the `value` is a correctly formatted scoreboard. Something like this

```python
if not all(c in self.SCOREBOARD_KEYS.keys() for c in value):
    self.log.info("Invalid scoreboard value, ignoring the scoreboard metrics")
    return
```

Reply:
> Additionally we might want to add a validation check at the beginning of that method to make sure that the value is a correctly formatted scoreboard. Something like this

I'm hesitant to put something like this in because it would completely stop reporting any of the new worker metrics when a newer version of Apache would add a new state. Personally I would prefer having the known metrics and only missing a new state when Apache decides to add one, instead of losing all the metrics I have added in this PR. Given that scoreboard is a core mechanism of Apache I can't imagine a day where one of there status boards will report a different kind of value under the name 'scoreboard'. So I would expect that the change of receiving a value that doesn't adhere to the expected format is lower than the change of them adding an extra state to the existing format. 

I don't have a strong opinion on this and would gladly put a check like that (probably exactly that check😉) in but this is why I haven't done it initially.

Reply:
The issue with Apache adding a new state was that your old code would raise an exception when trying to read from SCOREBOARD_KEYS, adding a pre-validation was a nice nit in that situation to give the user a more meaningful error message than a big stack trace.

By using `count`, we shouldn't raise any exception anymore. The code gracefully handles unknown character by ignoring it. Thus making the validation unneeded.

Reply:
Oh, off course. That's what you get from building in PHP on your day job 😅 , PHP doesn't care. I thought/hoped that sticking to the basics would prevent me from making those kinds of errors, but apparently it won't.